### PR TITLE
Update GitHub Actions to latest version

### PR DESCRIPTION
**Pull request by imjohnbo**
Hi Terraform team! 👋 

This PR contains updates the following actions to make them compliant with the [latest version](https://github.blog/2019-08-08-github-actions-now-supports-ci-cd/) of GitHub Actions:
- `terraform-apply`
- `terraform-fmt`
- `terraform-init`
- `terraform-plan`
- `terraform-validate`

In each case, the following changes were made:
- Moved metadata from Dockerfile to an `action.yml`
- Added contents of corresponding Terraform website documentation (ex. [`terraform-fmt`](https://www.terraform.io/docs/github-actions/actions/fmt.html)) to `README`
- Updated documentation to use new YAML syntax for workflows

For increased discoverability, GitHub recommends organizations' official actions to be on the [GitHub Marketplace](https://github.com/marketplace?type=actions). After splitting out these repositories into separate repositories and cutting a release for each one, it should be possible to publish to the Marketplace on the Releases tab. 🚀 

Happy to get your feedback, thanks for taking a look!

### CODE REVIEW

**Comment by sudomateo** [comment]
Thanks for this! I'm already working on refactoring these actions to one actions here: https://github.com/hashicorp/terraform-github-actions/tree/refactor-bash

**Comment by imjohnbo** [comment]
Very cool, @sudomateo! Looking forward to seeing that progress! 🚀

File: apply/action.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: 'Terraform Apply'
+description: 'Runs `terraform apply` and comments back on the pull request with the apply output.'
+author: 'HashiCorp Terraform Team <terraform@hashicorp.com>'
+branding:
+  icon: 'play-circle'  
+  color: 'purple'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```
Comment (sudomateo) [inline apply/action.yml]: Let's add back the trailing new line.

File: fmt/action.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: 'Terraform Fmt'
+description: 'Runs `terraform fmt` to validate all Terraform files in a directory are in the canonical format.
+ If any files differ, this action will comment back on the pull request with the diffs of each file.'
+author: 'HashiCorp Terraform Team <terraform@hashicorp.com>'
+branding:
+  icon: 'terminal'  
+  color: 'purple'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```
Comment (sudomateo) [inline fmt/action.yml]: Let's add back the trailing new line.

File: init/action.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: 'Terraform Init'
+description: 'Runs `terraform init` to initialize a Terraform working directory. This action will comment back on the pull request on failure.'
+author: 'HashiCorp Terraform Team <terraform@hashicorp.com>'
+branding:
+  icon: 'download'  
+  color: 'purple'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```
Comment (sudomateo) [inline init/action.yml]: Let's add back the trailing new line.

File: plan/action.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: 'Terraform Plan'
+description: 'Runs terraform plan and comments back on the pull request with the plan output.'
+author: 'HashiCorp Terraform Team <terraform@hashicorp.com>'
+branding:
+  icon: 'book-open'  
+  color: 'purple'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```
Comment (sudomateo) [inline plan/action.yml]: Let's add back the trailing new line.

File: validate/action.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+name: 'Terraform Validate'
+description: 'Runs terraform validate to validate the Terraform files in a directory comments back on error.'
+author: 'HashiCorp Terraform Team <terraform@hashicorp.com>'
+branding:
+  icon: 'alert-triangle'  
+  color: 'purple'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```
Comment (sudomateo) [inline validate/action.yml]: Let's add back the trailing new line.

**Comment by sudomateo** [comment]
Should be fixed in #88.

**Pull request closed without merging**